spec/mongoid_spec.rb

```diff
@@ -305,6 +310,33 @@ class SubExample < FreeFormExample
     expect(project.can_transition_to_state?(:retired)).to be_falsey
   end
 
+  describe 'tracking states' do
+    it 'should track changes to parent states' do
+      example.state = :archived
+      example.save
+      expect(example.inactive_value).to eq :archived
+      expect(example.inactive_at).to_not be_nil
+    end
+
+    it 'should track changes to child states' do
+      example.state = :published
+      example.save
+      expect(example.published_value).to eq :published
+      expect(example.published_at).to_not be_nil
+    end
+
+    it 'should create tracking properties for tracked states' do
+      expect(example).to respond_to(:published_at)
+      expect(example).to respond_to(:published_value)
```

Comment:
Should we have a published value field in this case? there is only one possible value that it could ever be. We should probably not bother with this field if its not a group thats being tracked.